tests/rptest/tests/follower_fetching_test.py

```diff
@@ -168,6 +168,76 @@ def test_basic_follower_fetching(self, read_from_object_store):
                 else:
                     assert current_bytes_fetched == 0
 
+    @cluster(num_nodes=5)
+    def test_follower_fetching_with_maintenance_mode(self):
+        rack_layout_str = "ABC"
+        rack_layout = [str(i) for i in rack_layout_str]
+
+        for ix, node in enumerate(self.redpanda.nodes):
+            extra_node_conf = {
+                # We're introducing two racks, small and large.
+                # The small rack has only one node and the
+                # large one has four nodes.
+                'rack': rack_layout[ix],
```

Comment:
Based on what I see, `rack_layour_str` would suffice and `rack_layout` is not needed.
```suggestion
                'rack': rack_layout_str[ix],
```